tests/lib/rules/prefer-presence-queries.test.ts

```diff
@@ -14,81 +14,329 @@ const queryAllByQueries = ALL_QUERIES_METHODS.map(
   (method) => `queryAll${method}`
 );
 
-const allQueryUseInAssertion = (queryName: string) => [
-  queryName,
-  `screen.${queryName}`,
-];
+type AssertionFnParams = {
+  query: string;
+  matcher: string;
+  messageId: MessageIds;
+  shouldUseScreen?: boolean;
+};
 
-const getValidAssertion = (query: string, matcher: string) =>
-  allQueryUseInAssertion(query).map((query) => ({
-    code: `expect(${query}('Hello'))${matcher}`,
-  }));
+const getValidAssertion = ({
+  query,
+  matcher,
+  shouldUseScreen = false,
+}: Omit<AssertionFnParams, 'messageId'>) => {
+  const finalQuery = shouldUseScreen ? `screen.${query}` : query;
+  return {
+    code: `expect(${finalQuery}('Hello'))${matcher}`,
+  };
+};
 
-const getInvalidAssertion = (
-  query: string,
-  matcher: string,
-  messageId: MessageIds
-) =>
-  allQueryUseInAssertion(query).map((query) => ({
-    code: `expect(${query}('Hello'))${matcher}`,
-    errors: [{ messageId }],
-  }));
+const getInvalidAssertion = ({
+  query,
+  matcher,
+  messageId,
+  shouldUseScreen = false,
+}: AssertionFnParams) => {
+  const finalQuery = shouldUseScreen ? `screen.${query}` : query;
+  return {
+    code: `expect(${finalQuery}('Hello'))${matcher}`,
+    errors: [{ messageId, line: 1, column: shouldUseScreen ? 15 : 8 }],
```

Comment:
Tests in this file are mostly the same, but I refactor the assertion makers to be able to check the column within the error.